src/main/ruby/truffleruby/core/struct.rb

```diff
@@ -154,8 +155,28 @@ def initialize(*args)
       raise ArgumentError, "Expected #{attrs.size}, got #{args.size}"
     end
 
-    attrs.each_with_index do |attr, i|
-      TrufflePrimitive.object_hidden_var_set self, attr, args[i]
+    if self.class::KEYWORD_INIT
+      if args.length > 1 || !args.first.is_a?(Hash)
```

Comment:
MRI seems to allow `Struct.new(:a,:b, keyword_init: true).new()`, i.e., 0 args.
Could you allow that too and add a spec?